addons/crm_livechat/models/chatbot_script_step.py

```diff
@@ -101,8 +101,9 @@ def _process_step_create_lead_and_forward(self, discuss_channel):
         users = discuss_channel.livechat_channel_id.sudo()._get_available_operators_by_livechat_channel(
             self.env["res.users"].browse(assignable_user_ids)
         )[discuss_channel.livechat_channel_id]
-        message = self._process_step_forward_operator(discuss_channel, users=users)
         operator_partner = discuss_channel.livechat_operator_id
+        message = self._process_step_forward_operator(discuss_channel, users=users)
+
         if operator_partner != self.chatbot_script_id.operator_partner_id:
```

Comment:
as discussed, the condition should become `if previous_operator != discuss_channel.livechat_operator_id:` for the change to make sense (we assign a user only if the operator changed, this is what the condition tried to achieve but it was making the wrong assumption about the bot rather than just actually checking if the operator changed)